### PR TITLE
Setup and Controllers

### PR DESCRIPTION
**Pull request by Lucas-Masaba**
# Project Requirements ⚙️ 

- Create a new app.
- Initialize project with Git.
- Set up Postgres database.
- For each URL, the following were created:
  - Route
  - Controller
  - View

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+  get 'users/index'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  root "users#index"
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show"
+  get "/users/:user_id/posts", to: "posts#index"
+  get "/users/:user_id/posts/:id", to: "posts#show"
```
Comment (Emmyn5600) [inline config/routes.rb]: Resource routing allows you to quickly declare all of the common routes for a given resourceful controller. A single call to resources can declare all of the necessary routes for your index , show , new , edit , create , update , and destroy actions.

- [x] Use the `resources` for the configuration in the routes like that:

```
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Lucas-Masaba 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [x] Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (Lucas-Masaba) [reply config/routes.rb]: Thank you so much for this tip. I will definitely work on it.

**Review by Emmyn5600 — APPROVED** [review]
Hi @Lucas-Masaba 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [x] N/A

#### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
#### _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**